app/assets/javascripts/modules/inline-attachment-modal.js

```diff
@@ -34,6 +34,9 @@ InlineAttachmentModal.prototype.actionCallback = function (item) {
     'insert': function () {
       this.workflow.render(window.ModalFetch.getLink(item))
     },
+    'confirm-delete': function () { 
```

Comment:
I'd actually switch this around with 'delete'. Thinking about it, the 'confirmation' is really on the next page, and the first action is to 'delete'. Does that make sense?